src/main/java/org/mastodon/views/PainterThread.java

```diff
@@ -0,0 +1,103 @@
+package org.mastodon.views;
+
+import java.lang.ref.WeakReference;
+import java.util.concurrent.RejectedExecutionException;
+
+import bdv.viewer.RequestRepaint;
+
+/**
+ * Thread to repaint display.
+ * <p>
+ * This is a copy of {@link bdv.viewer.render.PainterThread} with
+ * the significant difference that it uses a {@link WeakReference}
+ * to the {@link Paintable}. This has the following consequences:
+ * <ul>
+ *     <li>The thread never prevents the garbage collection of the paintable.</li>
+ *     <li>
+ *         The thread automatically shuts down if the paintable gets garbage collected.
+ *         It's not necessary (but still possible) to explicitly terminate the thread
+ *         by calling {@link Thread#interrupt() thread.interrupt()}.
+ *     </li>
+ * </ul>
+ */
+public class PainterThread extends Thread implements RequestRepaint
+{
+	public interface Paintable {
+
+		void paint();
+	}
+
+	private final WeakReference< Paintable > paintable;
+
+	private boolean pleaseRepaint;
+
+	public PainterThread( final Paintable paintable )
+	{
+		this( null, "PainterThread", paintable );
+	}
+
+	public PainterThread( final ThreadGroup group, final PainterThread.Paintable paintable )
+	{
+		this( group, "PainterThread", paintable );
+	}
+
+	public PainterThread( final ThreadGroup group, final String name, final Paintable paintable )
+	{
+		super( group, name );
+		this.paintable = new WeakReference<>( paintable );
+		this.pleaseRepaint = false;
+	}
```

Comment:
Maybe leave out these two constructors since they currently do not seem to be used in the project?